### PR TITLE
Implemented signers

### PR DESCRIPTION
**Pull request by belyakove**

### CODE REVIEW

**Review by popzxc — COMMENTED** [review]
Do all the dependencies have to be a part of the repository, or is it a `.gitignore` issue?

**Comment by belyakove** [comment]
> Do all the dependencies have to be a part of the repository, or is it a `.gitignore` issue?

They don't necessarily need to be in repository. They are there just to be able to run example app right away without need to run `pod install`. They can be removed from repo and put to `.gitignore` if needed. Please let me know how do you prefer

**Comment by popzxc** [comment]
I think less code in the repository is desirable, thus I'd suggest getting rid of it.

**Comment by belyakove** [comment]
> I think less code in the repository is desirable, thus I'd suggest getting rid of it.

Removed Pods from repo

**Comment by belyakove** [comment]
Can this one be merged?

**Comment by popzxc** [comment]
@StanislavBreadless Please review this PR and merge if it satisfies requirements.

**Review by StanislavBreadless — CHANGES_REQUESTED** [review]
Hi :slightly_smiling_face:, I have a few questions:

File: .gitignore [hunk 1]
```diff
@@ -34,4 +34,4 @@ Carthage/Build
 # Note: if you ignore the Pods directory, make sure to uncomment
 # `pod install` in .travis.yml
```
[the remark this answers is not in the document]
Comment (belyakove) [reply .gitignore:35]: This is only applicable if you are using github CI

File: ZKCryptoSDK/Classes/ZKCryptoSDK.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+import Foundation
+
+/**
+* Access to the ZksCrypto native library
+*
+* This class provides methods for interaction the ZksCrypto native library.
+*
+*/
+public class ZKCryptoSDK: NSObject {
```
[the remark this answers is not in the document]
Comment (belyakove) [reply ZKCryptoSDK/Classes/ZKCryptoSDK.swift:16]: It should be possible to use it as a separate pod. Was planning to look at the project structure as the final step

File: ZKSyncSDK/Classes/Signer/Utils/Utils.swift [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        let bits = Bits(dataBEOrder: decimalBytes).reversed
+        
+        var exponent = BigUInt.zero
+        var expPow2 = BigUInt.one
+        
+        for i in 0..<expBits {
+            exponent += bits[i] ? expPow2 : 0
+            expPow2 *= 2
+        }
+        exponent = BigUInt(expBase).power(Int(exponent))
```
[the remark this answers is not in the document]
Comment (belyakove) [reply ZKSyncSDK/Classes/Signer/Utils/Utils.swift:134]: Indeed Int can be used here. Will update

**Pull request closed without merging**